### ISSUE TITLE
What is the purpose of the leader election mechanism

### ISSUE THREAD

**Issue by sgandon** [kubernetes-sigs/nfs-subdir-external-provisioner#19]
Hello,
I have tried to use this nfs provisioner and it seems to respond to a lot of my needs especially the possilibity to have a single folder per PV(c) that matchs very well the multi-tenant architecture we have.
I have tried to install the helm chart with 2 replicas and from what I understand the second replica is waiting for the current leader to fail to answer requests.
What is the gain of having a second pod waiting for the first to fail comparing to a solution with a single pod restarting after a possible failure. 
Is it a matter of time availability ? because from what I have seen the restart seem to be even faster than the time the election occurs.

**Comment by docent-net** [comment]
Hi,

I'm not project maintainer. I found this question looking for some other answer ad your question made me think about it.

So looks like the leadership election is a feature of sigs.k8s.io/sig-storage-lib-external-provisioner/controller and defined here:

https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/blob/master/cmd/nfs-subdir-external-provisioner/provisioner.go#L266

So the answer to your question lays somewhere between RenewDeadline() here https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/controller/controller.go#L440 and retryPeriod() here: https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/controller/controller.go#L452

Summing up - IMO, in this case, I'm gonna go w/only one pod for this provisioner deployment, as the restarting process may be faster than default configuration of LeaderElection. 

Also looks like we can fine-tune the restart process of only 1 pod via deployment.yaml and standard k8s features such as Liveness/Readiness Probes.

**Comment by sgandon** [comment]
Hi,
Thanks for you answer, I could not agree more.
